app/models/sufia/permission_template.rb

```diff
@@ -26,19 +26,45 @@ def release_no_delay?
       release_period == RELEASE_TEXT_VALUE_NO_DELAY
     end
 
-    # Does this permission template require a date that all works are released before
+    # Does this permission template require a date (or embargo) that all works are released before
     # NOTE: date will be in release_date
     def release_before_date?
-      release_period == RELEASE_TEXT_VALUE_BEFORE_DATE
+      # All PermissionTemplate embargoes are dynamically determined release before dates
+      release_period == RELEASE_TEXT_VALUE_BEFORE_DATE || release_max_embargo?
     end
 
-    # Is a specific embargo period required by this permission template
-    # NOTE: embargo period will be in release_period
-    def release_embargo?
+    # Is there a maximum embargo period specified by this permission template
+    # NOTE: latest embargo date returned by release_date, maximum embargo period will be in release_period
+    def release_max_embargo?
```

Comment:
Nice, I like it. Will refactor, as I agree the current implementation is clunky.